shell/browser/api/atom_api_session.cc

```diff
@@ -646,6 +652,42 @@ void Session::Preconnect(const gin_helper::Dictionary& options,
                      url, num_sockets_to_preconnect));
 }
 
+#if BUILDFLAG(ENABLE_BUILTIN_SPELLCHECKER)
+base::Value Session::GetSpellCheckerLanguages() {
+  return browser_context_->prefs()
+      ->Get(spellcheck::prefs::kSpellCheckDictionaries)
+      ->Clone();
+}
+
+void Session::SetSpellCheckerLanguages(
+    gin_helper::ErrorThrower thrower,
+    const std::vector<std::string>& languages) {
+  base::ListValue language_codes;
+  for (const std::string& lang : languages) {
+    std::string code = spellcheck::GetCorrespondingSpellCheckLanguage(lang);
+    if (code.empty()) {
+      thrower.ThrowError("Invalid language code provided: \"" + lang +
+                         "\" is not a valid language code");
+      return;
+    }
+    language_codes.AppendString(code);
+  }
+  browser_context_->prefs()->Set(spellcheck::prefs::kSpellCheckDictionaries,
+                                 language_codes);
+}
+
+void SetSpellCheckerDictionaryDownloadURL(gin_helper::ErrorThrower thrower,
+                                          const GURL& url) {
+  if (!url.is_valid()) {
+    thrower.ThrowError(
+        "The URL you provided to setSpellCheckerDictionaryDownloadURL is not a "
+        "valid URL");
+    return;
+  }
+  SpellcheckHunspellDictionary::SetDownloadURLForTesting(url);
```

Comment:
Can we avoid using a testing api ? maybe get a new one from upstream ?

Reply:
Using testing was easy.  I can file a CL upstream to get a proper setter if we want 👍